management/src/test/java/org/ehcache/management/providers/statistics/StatsUtil.java

```diff
@@ -120,7 +120,7 @@ public static double getExpectedValueFromRatioHistory(String statName, Context c
         int mostRecentIndex = ratioHistory.getValue().length - 1;
         value = ratioHistory.getValue()[mostRecentIndex].getValue();
       }
-    } while (!Thread.currentThread().isInterrupted() && value != expectedResult);
+    } while (!Thread.currentThread().isInterrupted() && !value.equals(expectedResult));
```

Comment:
@ljacomet the sleep actually isn't in the loop.  I call sleep right after the get calls(allowing the stats time to be calculated) and before I enter the looping to gather the stat values.  Please let me know if I missed something but I am not sure how to implement your suggestion.